test/checks/label/label-content-name-mismatch.js

```diff
@@ -0,0 +1,127 @@
+describe('label-content-name-mismatch', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var check = undefined;
+	var options = undefined;
+	// var checkSetup = axe.testUtils.checkSetup;
+	// var shadowSupport = axe.testUtils.shadowSupport;
+
+	beforeEach(function() {
+		check = checks['label-content-name-mismatch'];
+	});
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+		check = undefined;
+		axe._tree = undefined;
+	});
+
+	it('returns false when element has an empty aria-label', function() {
+		fixture.innerHTML =
+			'<button id="target" name="link" aria-label="">The full label</button>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var vNode = axe.utils.getNodeFromTree(tree[0], node);
+		var actual = check.evaluate(node, options, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false when element has an empty aria-labelledby', function() {
+		fixture.innerHTML =
+			'<button id="target" name="link" aria-labelledby="">The full label</button>';
+		var node = fixture.querySelector('#target');
+		var tree = (axe._tree = axe.utils.getFlattenedTree(fixture));
+		var actual = check.evaluate(
+			node,
+			options,
+			axe.utils.getNodeFromTree(tree[0], node)
+		);
+		assert.isFalse(actual);
+	});
+
+	it('returns false when element has no text content', function() {
+		fixture.innerHTML =
+			'<button id="target" name="link" aria-label="Ok"></button>';
```

Comment:
This too, if there is no content, there is no mismatch and so this shouldn't fail.